pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -1,2 +1,221 @@
+    });
+  });
+
+  describe('update weights', () => {
+    sharedBeforeEach('deploy tokens', async () => {
+      const action = await actionId(pool.instance, 'updateWeightsGradually');
+      await pool.vault.grantRole(action, owner);
+    });
+
+    it('non-owners cannot update weights', async () => {
+      const now = await currentTimestamp();
+
+      await expect(pool.updateWeightsGradually(other, now, now, weights)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+
+    it('will fail if end weights do not match', async () => {
+      const now = await currentTimestamp();
+
+      await expect(pool.updateWeightsGradually(owner, now, now, weights.slice(0, 1))).to.be.revertedWith(
+        'INPUT_LENGTH_MISMATCH'
+      );
+    });
+
+    it('will succeed with start time in the past', async () => {
+      const now = await currentTimestamp();
+
+      await expect(pool.updateWeightsGradually(owner, now.sub(1000), now.add(1), weights)).not.to.be.reverted;
+    });
+
+    it('will fail if start time > end time', async () => {
+      const now = await currentTimestamp();
+
+      await expect(pool.updateWeightsGradually(owner, now, now.sub(1), weights)).to.be.revertedWith(
+        'GRADUAL_UPDATE_TIME_TRAVEL'
+      );
+    });
+
+    it('will fail with an invalid end weight', async () => {
+      const now = await currentTimestamp();
+      const badWeights = weights;
+      badWeights[2] = fp(0);
+
+      await expect(pool.updateWeightsGradually(owner, now.add(100), now.add(1000), badWeights)).to.be.revertedWith(
+        'MIN_WEIGHT'
+      );
+    });
+
+    it('will fail with invalid normalized end weights', async () => {
+      const now = await currentTimestamp();
+      const badWeights = Array(weights.length).fill(fp(0.6));
+
+      await expect(pool.updateWeightsGradually(owner, now.add(100), now.add(1000), badWeights)).to.be.revertedWith(
+        'NORMALIZED_WEIGHT_INVARIANT'
+      );
+    });
+
+    context('owner can update weights', () => {
+      // startWeights = [fp(0.3), fp(0.55), fp(0.1), fp(0.5)];
+      const endWeights = [fp(0.15), fp(0.25), fp(0.55), fp(0.05)];
+      const halfWeights = [fp(0.225), fp(0.4), fp(0.325), fp(0.05)];
+      let now, startTime: BigNumber, endTime: BigNumber;
+      const START_DELAY = MINUTE * 10;
+      const UPDATE_DURATION = MINUTE * 60;
+
+      sharedBeforeEach('updateWeightsGradually', async () => {
+        now = await currentTimestamp();
+        startTime = now.add(START_DELAY);
+        endTime = startTime.add(UPDATE_DURATION);
+
+        await pool.updateWeightsGradually(owner, startTime, endTime, endWeights);
+      });
+
+      it('stores the params', async () => {
+        const updateParams = await pool.getGradualWeightUpdateParams();
+
+        expect(updateParams.startTime).to.equalWithError(startTime, 0.001);
+        expect(updateParams.endTime).to.equalWithError(endTime, 0.001);
+        expect(updateParams.endWeights).to.equalWithError(endWeights, 0.001);
+      });
```

Comment:
We should have a test where start time is in the past, and it is fast-forwarded to now.